### PR TITLE
Remove pages ported to dcrdevdocs, add redirects

### PR DESCRIPTION
**Pull request by s-ben**
**NOTE: do not merge until developer documentation is launched.**

In preparation for dev docs launch, removing pages that have been moved to the new dev docs site, adding redirects.

A couple outstanding issues on this PR:

- Removed the #### Testnet reset section from the README, as it was pointing to a page that was moved. Put back in with the link to devdocs page?
- My naive attempt to do nginx redirects to another domain has failed…:/ Before spending more time on it, thought I’d see if anyone knew the solution.

### CODE REVIEW

File: conf/nginx.conf [hunk 1]
```diff
@@ -70,4 +70,9 @@ server {
     rewrite ^/proof-of-work/proof-of-work/?$                             $scheme://$http_host/proof-of-work/overview permanent;
     rewrite ^/governance/introduction-to-decred-governance/?$            $scheme://$http_host/governance/overview permanent;
     rewrite ^/contributing/contributing-to-decred/?$                     $scheme://$http_host/contributing/overview permanent;
+    rewrite ^/advanced/address-details/?$                                $https://devdocs.jamieholdstock.com/developer-guides/addresses permanent;
```
Comment (jholdstock) [inline conf/nginx.conf]: URLs need updating now that https://devdocs.decred.org is up

File: mkdocs.yml [hunk 2]
```diff
@@ -140,7 +134,6 @@ nav:
   - 'Script Extensions': 'research/script-extensions.md'
   - 'Signature Script Isolation and Fraud Proofs': 'research/signature-script-isolation-and-fraud-proofs.md'
   - 'Block Production Times': 'research/block-production-times.md'
-  - 'Transaction Extensions': 'research/transaction-extensions.md'
```
Comment (jholdstock) [inline mkdocs.yml]: No redirect for this

File: mkdocs.yml [hunk 3]
```diff
@@ -113,19 +113,13 @@ nav:
     - 'Voting Tickets': 'faq/proof-of-stake/voting-tickets.md'
     - 'Voting Service Providers': 'faq/proof-of-stake/voting-service-providers.md'
 - Advanced:
-  - 'Address Details': advanced/address-details.md
-  - 'Using Testnet': 'advanced/using-testnet.md'
   - 'Manual CLI Installation': 'advanced/manual-cli-install.md'
-  - 'Block Header Specifications': 'advanced/block-header-specifications.md'
```
Comment (jholdstock) [inline mkdocs.yml]: No redirect for this

File: docs/research/overview.md [hunk 4]
```diff
@@ -16,7 +16,6 @@ Research work in Decred is currently organized around the following subsections:
 * [BLAKE-256 Hash Function](blake-256-hash-function.md)
 * [Script Extensions](script-extensions.md)
 * [Signature Script Isolation and Fraud Proofs](signature-script-isolation-and-fraud-proofs.md)
-* [Transaction Extensions](transaction-extensions.md)
```
Comment (jholdstock) [inline docs/research/overview.md]: Link to devdocs instead of removing link completely?

File: docs/faq/wallets-and-seeds.md [hunk 5]
```diff
@@ -64,7 +64,7 @@ dcrctl --wallet getbalance "imported" 0 all
 
 #### 7. What is the difference between a testnet and mainnet public key address?
 
-A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with its 2-byte prefix which identifies the network and type. A mainnet public key address starts with the letters `Dk` while a testnet public key address starts with the letters `Tk`. For details on address types, see the see the [address details](../advanced/address-details.md) page.  
+A public key address, also called Pay-To-Pubkey (P2Pk), can be identified with its 2-byte prefix which identifies the network and type. A mainnet public key address starts with the letters `Dk` while a testnet public key address starts with the letters `Tk`. For details on address types, see the see the [address details](https://devdocs.decred.org/developer-guides/addresses/) page in the developer documentation.
```
Comment (jholdstock) [inline docs/faq/wallets-and-seeds.md]: `see the see the`

Comment (s-ben) [reply docs/faq/wallets-and-seeds.md]: Fixed

File: docs/research/overview.md (hunk 4 above)
Comment (s-ben) [reply docs/research/overview.md]: Good idea. Linking to devdocs now

File: mkdocs.yml (hunk 3 above)
Comment (s-ben) [reply mkdocs.yml]: Added redirect, though not sure if working...

File: mkdocs.yml (hunk 2 above)
Comment (s-ben) [reply mkdocs.yml]: Added redirect, though not sure if working...

File: conf/nginx.conf (hunk 1 above)
Comment (s-ben) [reply conf/nginx.conf]: Done

**Comment by s-ben** [comment]
@jholdstock have addressed comments. The only thing I'm (potentially) stuck on is the last two redirects added:

rewrite ^/advanced/block-header-specifications/?$                    https://devdocs.decred.org/developer-guides/block-header-specifications     permanent;

rewrite ^/research/transaction-extensions/?$                         https://devdocs.decred.org/developer-guides/transactions/transaction-extensions/     permanent;

These might be working, but locally redirects aren't working for some reason, even though I've managed to get the links matching what's in the redirects...:/ Starting to spin my wheels here, so thought I'd see if you saw the problem right away.

File: docs/research/ticket-selection-algorithm.md [hunk 6]
```diff
@@ -4,5 +4,5 @@ Every block, 5 tickets are pseudorandomly selected from the ticket pool to [vote
...
+Tickets are selected from the ticket pool using a deterministic [Pseudorandom Number Generator (PRNG)](https://en.wikipedia.org/wiki/Pseudorandom_number_generator). It is deterministic because the seed input to the PRNG is derived from the hash of the previous block (the block being voted on). Specifically, the seed is a serialization of the [header](/advanced/block-header-specifications/) of the previous block suffixed with a constant derived from the hex representation of the mathematical constant Pi (𝜋), which acts as a publicly verifiable Nothing-Up-My-Sleeve ([NUMS](https://en.wikipedia.org/wiki/Nothing-up-my-sleeve_number)) constant. All eligible tickets in the ticket pool (also referred to as live tickets) are hashed using the [BLAKE-256](blake-256-hash-function.md) hash function and sorted [lexicographically](https://en.wikipedia.org/wiki/Lexicographical_order) by their hash to generate a [total order](https://en.wikipedia.org/wiki/Total_order). Finally, uniformly random values constrained by the number of total eligible tickets as an upper bound are produced and used as indices into the total order.
```
Comment (jholdstock) [inline docs/research/ticket-selection-algorithm.md]:
```
[header](/advanced/block-header-specifications/)
```

This page is now in devdocs.

File: docs/wallets/spv.md [hunk 7]
```diff
@@ -18,7 +18,7 @@ Another benefit offered by SPV is an extreme reduction in the time required for
 
 ## How does SPV work?
 
-At the start of every block added to the Decred blockchain is 180 bytes of data called the [block header](../advanced/block-header-specifications.md). The block header describes key information about the block including the hash of the block, the Merkle root (the sum of all the transaction hashes in the block), and the nonce calculated by the proof-of-work miners. A predetermined filter is also created for every block, based on the all of the transactions within the block.
+At the start of every block added to the Decred blockchain is 180 bytes of data called the [block header](../../advanced/block-header-specifications/). The block header describes key information about the block including the hash of the block, the Merkle root (the sum of all the transaction hashes in the block), and the nonce calculated by the proof-of-work miners. A predetermined filter is also created for every block, based on the all of the transactions within the block.
```
Comment (jholdstock) [inline docs/wallets/spv.md]:
```
[block header](../../advanced/block-header-specifications/)
```

This page is now in devdocs.

**Comment by jholdstock** [comment]
Redirections look fine to me. Here is how I have tested them:

```
# Build docker container
$ bin/build.sh

# Run docker container
$ docker run -d --rm -p 8000:80 decred/dcrdocs:latest

# cURL and output HTTP headers
$ curl -I http://localhost:8000/advanced/block-header-specifications/
```

Here's the output - see the value of the `Location:` header:

```
HTTP/1.1 301 Moved Permanently
Server: nginx
Date: Sat, 02 Nov 2019 10:27:26 GMT
Content-Type: text/html
Content-Length: 162
Connection: keep-alive
Location: https://devdocs.decred.org/developer-guides/block-header-specifications
Expires: Thu, 01 Jan 1970 00:00:01 GMT
Cache-Control: no-cache
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-XSS-Protection: 1; mode=block
Referrer-Policy: no-referrer
```

File: docs/research/ticket-selection-algorithm.md (hunk 6 above)
Comment (s-ben) [reply docs/research/ticket-selection-algorithm.md]: Fixed

File: docs/wallets/spv.md (hunk 7 above)
Comment (s-ben) [reply docs/wallets/spv.md]: Fixed

**Pull request merged**